src/services/battleye/battleye_service.cpp

```diff
@@ -0,0 +1,247 @@
+			*p++ = sEncodingTable[((data[i + 1] & 0xF) << 2) | ((int)(data[i + 2] & 0xC0) >> 6)];
+			*p++ = sEncodingTable[data[i + 2] & 0x3F];
+		}
+		if (i < in_len)
+		{
+			*p++ = sEncodingTable[(data[i] >> 2) & 0x3F];
+			if (i == (in_len - 1))
+			{
+				*p++ = sEncodingTable[((data[i] & 0x3) << 4)];
+				*p++ = '=';
+			}
+			else
+			{
+				*p++ = sEncodingTable[((data[i] & 0x3) << 4) | ((int)(data[i + 1] & 0xF0) >> 4)];
+				*p++ = sEncodingTable[((data[i + 1] & 0xF) << 2)];
+			}
+			*p++ = '=';
+		}
+
+		return ret;
+	}
+
+}
+
+namespace big
+{
+	class battleye_server : looped_command
+	{
+		using looped_command::looped_command;
+
+		virtual void on_enable() override
+		{
+			g_battleye_service.start();
+		}
+
+		virtual void on_tick() override
+		{
+		}
+
+		virtual void on_disable() override
+		{
+			g_battleye_service.stop();
+		}
+	};
+
+	battleye_server g_battleye_server("battleyeserver", "BATTLEYE_SERVER", "BATTLEYE_SERVER_DESC", g.debug.battleye_server);
+
+	bool battleye_service::is_running()
+	{
+		return m_battleye_api.m_shutdown != nullptr;
+	}
+
+	void battleye_service::send_message(std::uint64_t token, void* message, int size)
+	{
+		packet pkt;
+
+		char header_buf[32];
+		rage::datBitBuffer header(header_buf, sizeof(header_buf));
+
+		header.Write<int>(size, 11);
+		header.Write<bool>(false, 1); // we are not the client
+
+		pkt.write_message(rage::eNetMessage::MsgBattlEyeCmd);
+		pkt.m_buffer.WriteArray(&header_buf, header.GetDataLength() * 8);
+		pkt.m_buffer.WriteArray(message, size * 8);
+
+		// send to player
+		if (auto plyr = g_player_service->get_by_host_token(token); plyr && plyr->get_session_player())
+		{
+			pkt.send(plyr->get_session_player()->m_msg_id, true);
+		}
+	}
+
+	void battleye_service::kick_player(std::uint64_t token, const char* reason)
+	{
+		if (auto plyr = g_player_service->get_by_host_token(token))
+		{
+			LOGF(WARNING, "BattlEye wants us to kick {} for {}", plyr->get_name(), reason);
+			if (!plyr->tampered_with_be)
+				session::add_infraction(plyr, Infraction::TRIGGERED_ANTICHEAT);
+		}
+	}
+
+	void battleye_service::script_func()
+	{
+		bool was_host = false;
+		while (is_running())
+		{
+			if (g_player_service->get_self()->is_valid())
+			{
+				bool is_host = g_player_service->get_self()->is_host();
+
+				if (is_host != was_host)
+				{
+					if (is_host)
+					{
+						for (auto& [_, player] : g_player_service->players())
+						{
+							add_player(player->get_net_game_player()->get_host_token(), player->get_rockstar_id(), player->get_name());
+						}
+					}
+					was_host = is_host;
+				}
+			}
+
+			big::script::get_current()->yield();
+		}
+	}
+
+	void battleye_service::thread_func()
+	{
+		while (is_running() && g_running)
+		{
+			// we need to run this every frame (lockstep with game or otherwise)
+			{
+				std::lock_guard lock(m_mutex);
+				if (is_running() && !m_battleye_api.m_run())
+					LOG(WARNING) << "BE::Run failed";

```

Comment:
Encompass in curly braces.

Reply:
Though it makes it much more readable, somewhere in 2022 they introduced the [InsertBraces](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#insertbraces) rule.

I'm for adding this/changing this rule in the clang-format file.